### PR TITLE
 .well-known support #2982

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Require https://github.com/matrix-org/matrix-android-sdk/pull/436

Autodiscovery is performed when the userLogin field loses focus and when the field value is a MX id (@xxx:<domain>)

### CODE REVIEW

File: vector/src/main/java/im/vector/activity/LoginActivity.java [hunk 1]
```diff
@@ -672,6 +675,86 @@ public void afterTextChanged(Editable s) {
 
         mTextInputLayouts = ViewUtilKt.findAllTextInputLayout(mFormContainer);
         ViewUtilKt.autoResetTextInputLayoutErrors(mTextInputLayouts);
+
+
+        mLoginEmailTextView.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) return;
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/LoginActivity.java]: I'm wondering if this is called when we switch to create account mode (?)

File: vector/src/main/java/im/vector/activity/LoginActivity.java [hunk 2]
```diff
@@ -339,6 +340,8 @@
     // use to reset the password when the user click on the email validation
     private ThreePidCredentials mForgotPid = null;
 
+    Boolean shouldAutodiscoverDomainBasedOnMail = false;
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/LoginActivity.java]: I would remove this code and all related to this.

File: vector/src/main/java/im/vector/activity/LoginActivity.java [hunk 3]
```diff
@@ -672,6 +675,86 @@ public void afterTextChanged(Editable s) {
...
         mTextInputLayouts = ViewUtilKt.findAllTextInputLayout(mFormContainer);
         ViewUtilKt.autoResetTextInputLayoutErrors(mTextInputLayouts);
+
+
+        mLoginEmailTextView.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) return;
+            //Maybe debounce
+            String candidate = mLoginEmailTextView.getText().toString();
+            //Try to see if we can find a domain
+            if (!candidate.isEmpty() && candidate.startsWith("@") && candidate.contains(":")) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/LoginActivity.java]: Use `MXPatterns.isUserId()`

File: vector/src/main/java/im/vector/activity/LoginActivity.java (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/activity/LoginActivity.java]: Tested, it's not called (it's another text view for register)

File: vector/src/main/java/im/vector/activity/LoginActivity.java (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/activity/LoginActivity.java]: ok

File: vector/src/main/java/im/vector/activity/LoginActivity.java (hunk 3 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/activity/LoginActivity.java]: done

File: vector/src/main/java/im/vector/activity/LoginActivity.java [hunk 4]
```diff
@@ -672,6 +674,87 @@ public void afterTextChanged(Editable s) {
...
         mTextInputLayouts = ViewUtilKt.findAllTextInputLayout(mFormContainer);
         ViewUtilKt.autoResetTextInputLayoutErrors(mTextInputLayouts);
+
+
+        mLoginEmailTextView.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) return;
+            //Maybe debounce
+            String candidate = mLoginEmailTextView.getText().toString();
+            //Try to see if we can find a domain
+            if (!candidate.isEmpty() && MXPatterns.isUserId(candidate)) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/LoginActivity.java]: canditate.isEmpty is not necessary, isUserId will return false

**Comment by BillCarsonFr** [comment]
:/
autcomplete

File: vector/src/main/java/im/vector/activity/LoginActivity.java (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/LoginActivity.java]: ok

**Pull request merged**